### PR TITLE
fix: x11 preview show without content

### PR DESCRIPTION
**Pull request by tsic404**
paint event get new preview content frequently make kwin a high cpu make a timer show preview 500ms later

log: as title

### CODE REVIEW

**Review by wineee — APPROVED** [review]

File: panels/dock/taskmanager/x11preview.cpp [hunk 1]
```diff
@@ -95,6 +99,67 @@ class AppItemWindowModel : public QAbstractListModel
...
+        QVariantMap option;
+        option["include-decoration"] = true;
+        option["include-cursor"] = false;
+        option["native-resolution"] = true;
+        args << QVariant::fromValue(option);
+        // 第三个参数，文件描述符
+        args << QVariant::fromValue(QDBusUnixFileDescriptor(fd[1]));
+
+        QDBusReply<QVariantMap> reply = interface.callWithArgumentList(QDBus::Block, QStringLiteral("CaptureWindow"), args);
+        if(!reply.isValid()) {
```
Comment (zccrs) [inline panels/dock/taskmanager/x11preview.cpp:130]: `if (`

File: panels/dock/taskmanager/x11preview.cpp [hunk 2]
```diff
@@ -95,6 +99,67 @@ class AppItemWindowModel : public QAbstractListModel
...
+        QVariantMap imageInfo = reply.value();
+        int imageWidth = imageInfo.value("width").toUInt();
+        int imageHeight = imageInfo.value("height").toUInt();
+        int imageStride = imageInfo.value("stride").toUInt();
+        int imageFormat = imageInfo.value("format").toUInt();
+
+        QFile file;
+        if (!file.open(fd[0], QIODevice::ReadOnly)) {
+            file.close();
+            ::close(fd[0]);
```
Comment (zccrs) [inline panels/dock/taskmanager/x11preview.cpp:149]: fd[1]不需要处理吗

File: panels/dock/taskmanager/x11preview.cpp [hunk 3]
```diff
@@ -95,6 +99,67 @@ class AppItemWindowModel : public QAbstractListModel
...
+            return QPixmap();
+        }
+
+        QImage::Format qimageFormat = static_cast<QImage::Format>(imageFormat);
+        int bitsCountPerPixel = QImage::toPixelFormat(qimageFormat).bitsPerPixel();
+
+        QByteArray fileContent = file.read(imageHeight * imageWidth * bitsCountPerPixel / 8);
+        QImage image(reinterpret_cast<uchar *>(fileContent.data()), imageWidth, imageHeight, imageStride, qimageFormat);
+        // close read
+        ::close(fd[0]);
```
Comment (zccrs) [inline panels/dock/taskmanager/x11preview.cpp:159]: fd[1]不需要处理吗

**Review by zccrs — CHANGES_REQUESTED** [review]

File: panels/dock/taskmanager/x11preview.cpp (hunk 2 above)
Comment (tsic404) [reply panels/dock/taskmanager/x11preview.cpp:149]: fd[1] 在call完dbus后就处理了

File: panels/dock/taskmanager/x11preview.cpp (hunk 3 above)
Comment (tsic404) [reply panels/dock/taskmanager/x11preview.cpp:159]: fd[1] 在call完dbus后就处理了

**Review by zccrs — APPROVED** [review]

**Pull request merged**